include/engine/api/route_parameters.hpp

```diff
@@ -94,20 +104,54 @@ struct RouteParameters : public BaseParameters
                     const boost::optional<bool> continue_straight_,
                     Args... args_)
         : BaseParameters{std::forward<Args>(args_)...}, steps{steps_}, alternatives{alternatives_},
-          annotations{annotations_}, geometries{geometries_}, overview{overview_},
-          continue_straight{continue_straight_}
+          annotations{annotations_},
+          annotations_type{annotations_ ? AnnotationsType::All : AnnotationsType::None},
+          geometries{geometries_}, overview{overview_}, continue_straight{continue_straight_}
+    {
+    }
+
+    // enum based implementation of annotations parameter
+    template <typename... Args>
+    RouteParameters(const bool steps_,
+                    const bool alternatives_,
+                    const AnnotationsType annotations_,
+                    const GeometriesType geometries_,
+                    const OverviewType overview_,
+                    const boost::optional<bool> continue_straight_,
+                    Args... args_)
+        : BaseParameters{std::forward<Args>(args_)...}, steps{steps_}, alternatives{alternatives_},
+          annotations_type{annotations_}, annotations{true}, geometries{geometries_},
+          overview{overview_}, continue_straight{continue_straight_}
     {
     }
 
     bool steps = false;
     bool alternatives = false;
     bool annotations = false;
```

Comment:
this is now a duplication, better
```
    bool annotations() const { return annotations_type != AnnotationsType::None; };
```

But it is not used in code base anymore, the best would be delete it and update unit tests.